### PR TITLE
WT-8485 Revert JSON messages for wt_progress

### PR DESCRIPTION
**Pull request by alisonfel**
Upon further evaluation, we've unfortunately misinterpreted the
purpose of the progress interface, where the message passed via the
interface should just be an operation string. Hence there is no
requirement for this string to be a complex message string.
This reverts changes introduced in WT-8396. However manually
reverts changes only relating to '__wt_progress' (as there are some
other general improvements worth keeping in the original commit).

### CODE REVIEW

File: src/include/connection.h [hunk 1]
```diff
@@ -560,7 +560,6 @@ struct __wt_connection_impl {
 /* AUTOMATIC FLAG VALUE GENERATION START 0 */
 #define WT_JSON_OUTPUT_ERROR 0x1u
 #define WT_JSON_OUTPUT_MESSAGE 0x2u
-#define WT_JSON_OUTPUT_PROGRESS 0x4u
     /* AUTOMATIC FLAG VALUE GENERATION STOP 8 */
```
Comment (etienneptl) [inline src/include/connection.h:563]: Shouldn't this line be updated too after running `s_all`?

Comment (alisonfel) [reply src/include/connection.h:563]: Hmm, I don't follow. After removing `WT_JSON_OUTPUT_PROGRESS`, `s_all` isn't updating anything further here.

Comment (etienneptl) [reply src/include/connection.h:563]: Sorry my bad, I don't why I thought this should be updated by `s_all`.

**Review by etienneptl — APPROVED** [review]
LGTM

**Pull request merged**